modules/gdscript/gdscript_compiler.cpp

```diff
@@ -241,9 +241,9 @@ static bool _can_use_validate_call(const MethodBind *p_method, const Vector<GDSc
 	}
 	MethodInfo info;
 	ClassDB::get_method_info(p_method->get_instance_class(), p_method->get_name(), &info);
-	for (int i = 0; i < p_arguments.size(); i++) {
-		const PropertyInfo &prop = info.arguments[i];
-		if (!_is_exact_type(prop, p_arguments[i].type)) {
+	int i = 0;
+	for (List<PropertyInfo>::ConstIterator itr = info.arguments.begin(); itr != info.arguments.end(); ++itr, ++i) {
```

Comment:
There's a few of these cases, seems a common paradigm.

I do wonder if using the `for each` simpler version and putting the `i++` on the first line might end up being easier to read. Maybe there's a more elegant way.

Reply:
Can look into them

Reply:
I did these, as seen above, to avoid any risky cases to ensure the iteration is clean, keeping the iteration in one place feels safer to me given some errors I accidentally caused with some of these cases when I used `for each` syntax, so will keep these for now for a potential cleanup down the line to ensure they work correctly